cachi2/core/package_managers/bundler/main.py

```diff
@@ -26,3 +31,14 @@ def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) ->
     """Process a request for a single bundler package."""
     parse_lockfile(package_dir)
     return []
+
+
+def _prepare_environment_variables_for_hermetic_build() -> list[EnvironmentVariable]:
+    # NOTE: these variables are necessary for hermetic builds and have to be set for
+    # the entire repository. There is a small chance that their values will be
+    # overridden by values in .bundle/configs checked in some of packages' repositories.
```

Comment:
Should we check at first if `BUNDLE_APP_CONFIG` is set and then use default location ?

Reply:
My understanding is that if a variable is set in environment and is not set in any config then it will be taken from the environment so there is no need to use default location. Here are results of a quick experiment I just ran:
```
$ bundle init
Writing new Gemfile to /tmp/bc/Gemfile
$ ls -a
.  ..  Gemfile
$ bundle config
Settings are listed in order of priority. The top value will be used.
$ export BUNDLE_NO_PRUNE=true
$ bundle config
Settings are listed in order of priority. The top value will be used.
no_prune
Set via BUNDLE_NO_PRUNE: true
$ bundle config local.BUNDLE_RETRY 25
$ bundle config
Settings are listed in order of priority. The top value will be used.
local.bundle_retry
Set for your local app (/tmp/bc/.bundle/config): "25"

no_prune
Set via BUNDLE_NO_PRUNE: true

$ ls -a
.  ..  .bundle  Gemfile
$ cat .bundle/config 
---
BUNDLE_LOCAL__BUNDLE_RETRY: "25"
$ sed -i s/25/14/g .bundle/config
$ bundle config
Settings are listed in order of priority. The top value will be used.
local.bundle_retry
Set for your local app (/tmp/bc/.bundle/config): "14"

no_prune
Set via BUNDLE_NO_PRUNE: true

$ sed -i s/BUNDLE_LOCAL__BUNDLE_RETRY/BUNDLE_RETRY/g .bundle/config
$ bundle config
Settings are listed in order of priority. The top value will be used.
no_prune
Set via BUNDLE_NO_PRUNE: true

retry
Set for your local app (/tmp/bc/.bundle/config): 14
```

It looks like as long as there is no clash between local and environment we are good to set whatever we need in the environment. Judging from my yesterdays search on Github the probability of a clash is very small.

Reply:
Yes, this makes total sense, thank you. I'll switch to using a dedicated config and a variable to point to it in the next revision.